File: atcoder/abc170B_crane_and_turtle.nim
# uninhm
# https://atcoder.jp/contests/abc170/tasks/abc170_b
# implementation

from strutils import split, parseInt
from sequtils import map

var x, y: int
(x, y) = stdin.readLine.split.map(parseInt)

if y mod 2 == 0 and y <= 4*x and y >= 2*x:
  echo "Yes"
else:
  echo "No"

File: atcoder/abc170B_crane_and_turtle.ml
(* vicfred
 * https://atcoder.jp/contests/abc170/tasks/abc170_b
 * brute force *)
let solve x y =
    let a, b = 4*x - y, y - 2*x in
    if a mod 2 = 0 && b mod 2 = 0 && a >= 0 && b >= 0 then "Yes" else "No"

let main =
    Scanf.scanf "%d %d" solve |> Printf.printf "%s\n"

